packages/core/src/plugins/logStore/network/NetworkProver.ts

```diff
@@ -0,0 +1,23 @@
+import { BroadbandPublisher } from '../../../shared/BroadbandPublisher';
+import { TlsProof } from '../protobuff/generated/prover';
+import { Prover } from '../Prover';
+
+export class NetworkProver extends Prover {
+	async handleNewProof(proof: TlsProof) {
+		this.logger.info(`A new proof with id: ${proof.id} has been receieved`);
+		try {
+			const { stream: streamId } = proof;
+			const streamrStream = await this.streamrClient.getStream(streamId);
+
+			const publisher = new BroadbandPublisher(
+				this.streamrClient,
+				streamrStream
+			);
+			await publisher.publish(proof);
```

Comment:
**A note**
We cannot just public the proof. We also need to include a reference to the Network Node gateway or ECDSA publisher/public key that notarised the request to produce the TlsProof.
This way the Notary Public Key/Cert can be downloaded by the verifier.